### PR TITLE
Feature/#258 log error stack trace

### PR DESCRIPTION
**Pull request by StarpTech**
```js
fastify
  .get('/', schema, function (req, reply) {
    reply
      .send(Promise.reject(new Error('deded')))
  })
```

### CODE REVIEW

**Comment by mcollina** [comment]
IMHO we can do the same thing by adding the standard error serializer, and not logging them in two lines.

Ideally the `error`  should be logged as `err` property, not `error`.


https://github.com/pinojs/pino/blob/master/docs/API.md#errSerializer

**Comment by allevo** [comment]
Hi,
  you can configure pino to serialize errors. See [here](http://getpino.io/#/docs/API?id=parameters) and [here](https://github.com/fastify/fastify/blob/master/lib/logger.js#L40)

The options used to create the pino instance is provided by [fastify constructor](https://github.com/fastify/fastify/blob/master/fastify.js#L36). So you can do this right now!

Anyway, I agree with you: we need to figure out how the stack should be logged

**Comment by mcollina** [comment]
> Anyway, I agree with you: we need to figure out how the stack should be logged

pino-http added the `err` serializer but when we did #170 we forgot. That's the fix.

**Comment by StarpTech** [comment]
@mcollina did you mean this? I updated

File: lib/logger.js [hunk 1]
```diff
@@ -47,7 +47,19 @@ const serializers = {
...
+      message: err.message,
+      stack: err.stack
+    }
+    for (var key in err) {
+      if (obj[key] === undefined) {
+        obj[key] = err[key]
+      }
+    }
+    return obj
+  },
```
Comment (mcollina) [inline lib/logger.js]: you can use `pino.stdSerializers.err`

Comment (StarpTech) [reply lib/logger.js]: I wanted to use it but it creates a hard binding to the module.

Comment (mcollina) [reply lib/logger.js]: this same file is already requiring it: https://github.com/StarpTech/fastify/blob/e1bde9c123165b03cd372ceddf97d38d122c59da/lib/logger.js#L9.

Comment (StarpTech) [reply lib/logger.js]: Fixed

**Pull request merged**